### PR TITLE
docs: guide to manually publish docker images

### PR DESCRIPTION
**Pull request by rootulp**
Motivated by https://github.com/celestiaorg/celestia-app/issues/3934

### CODE REVIEW

File: docs/maintainers/docker.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Docker
+
+## Context
+
+Github Actions should automatically build and publish a Docker image for each release. If Github Actions failed to build or publish the Docker image, you can manually build and publish a Docker image using this guide.
```
Comment (ninabarbakadze) [inline docs/maintainers/docker.md:5]: I think we could condense this

**Review by cmwaters — APPROVED** [review]
Would this also push the tx sim image?

File: docs/maintainers/docker.md (hunk 1 above)
Comment (rootulp) [reply docs/maintainers/docker.md:5]: Do you have a suggestion? Would you prefer this?

```suggestion
Github Actions should automatically build and publish a Docker image for each release. If Github Actions failed, you can manually build and publish a Docker image using this guide.
```

**Comment by rootulp** [comment]
The steps currently publish just the celestia-app Docker image b/c

```
docker buildx build --platform linux/amd64,linux/arm64 -t ghcr.io/celestiaorg/celestia-app:v2.2.0-mocha --push .
```

but I think this command could be slightly modified to build the txsim docker image and push it

**Review by rach-id — APPROVED** [review]

**Review by ninabarbakadze — APPROVED** [review]

**Pull request merged**